src/core/markdown.js

```diff
@@ -182,118 +176,6 @@ function workaroundBlockLevelMarkdown(element, selector) {
   }
 }
 
-class Builder {
-  constructor(doc) {
-    this.doc = doc;
-    this.root = doc.createDocumentFragment();
-    this.stack = [this.root];
-    this.current = this.root;
-  }
-  findPosition(header) {
-    return parseInt(header.tagName.charAt(1), 10);
-  }
-  findParent(position) {
-    let parent;
-    while (position > 0) {
-      position--;
-      parent = this.stack[position];
-      if (parent) return parent;
-    }
-  }
-  findHeader({ firstChild: node }) {
-    while (node) {
-      if (/H[1-6]/.test(node.tagName)) {
-        return node;
-      }
-      node = node.nextSibling;
-    }
-    return null;
-  }
-
-  addHeader(header) {
-    const section = this.doc.createElement("section");
-    const position = this.findPosition(header);
-
-    section.appendChild(header);
-    this.findParent(position).appendChild(section);
-    this.stack[position] = section;
-    this.stack.length = position + 1;
-    this.current = section;
-  }
-
-  addSection(node, process) {
-    const header = this.findHeader(node);
-    const position = header ? this.findPosition(header) : 1;
-    const parent = this.findParent(position);
-
-    if (header) {
-      node.removeChild(header);
-    }
-
-    node.appendChild(process(node));
-
-    if (header) {
-      node.prepend(header);
-    }
-
-    parent.appendChild(node);
-    this.current = parent;
-  }
-
-  addElement(node) {
-    this.current.appendChild(node);
-  }
-}
-
-function structure(fragment, doc) {
-  function process(root) {
-    const stack = new Builder(doc);
-    while (root.firstChild) {
-      const node = root.firstChild;
-      if (node.nodeType !== Node.ELEMENT_NODE) {
-        root.removeChild(node);
-        continue;
-      }
```

Comment:
This is the only thing I had to remove... it was discarding too much stuff (well, stuff that was causing tests to fail that were expecting comment nodes... but I think it also trashed TEXT nodes too).